src/network/networkd-address-label.c

```diff
@@ -169,43 +169,14 @@ int config_parse_address_label_prefix(const char *unit,
         if (r < 0)
                 return r;
 
-        /* AddressLabel=prefix/prefixlen */
-
-        /* prefixlen */
-        e = strchr(rvalue, '/');
-        if (e) {
-                unsigned i;
-
-                r = safe_atou(e + 1, &i);
-                if (r < 0) {
-                        log_syntax(unit, LOG_ERR, filename, line, r, "Prefix length is invalid, ignoring assignment: %s", e + 1);
-                        return 0;
-                }
-
-                if (i > 128) {
-                        log_syntax(unit, LOG_ERR, filename, line, r, "Prefix length is out of range, ignoring assignment: %s", e + 1);
-                        return 0;
-                }
-
-                n->prefixlen = (unsigned char) i;
-
-                prefix = strndupa(rvalue, e - rvalue);
-        } else
-                prefix = rvalue;
-
-        r = in_addr_from_string_auto(prefix, &f, &buffer);
+        r = in_addr_prefix_from_string(rvalue, AF_INET6, &buffer, &prefixlen);
         if (r < 0) {
-                log_syntax(unit, LOG_ERR, filename, line, r, "Address label is invalid, ignoring assignment: %s", prefix);
-                return 0;
-        }
-
-        if (f != AF_INET6) {
-                log_syntax(unit, LOG_ERR, filename, line, 0, "Address label family is not IPv6, ignoring assignment: %s", prefix);
+                log_syntax(unit, LOG_ERR, filename, line, r, "Address label is invalid, ignoring assignment: %s", rvalue);
                 return 0;
         }
 
-        n->family = f;
         n->in_addr = buffer;
+        n->prefixlen = prefixlen;
```

Comment:
You don't need `buffer` and `prefixlen` in this function. The way that `in_addr_prefix_from_string` is written, it will set the two output variables on success only. So you can pass `&n->in_addr` and `&n->prefixlen` directly.